pageserver/src/tenant.rs

```diff
@@ -350,9 +361,604 @@ impl Drop for TimelineUninitMark {
+                    // FIXME using None is a hack, it wont hurt, just ugly.
+                    //     Ideally initialize_with_lock error should return timeline in the error
+                    //     Or return ownership of itself completely so somethin like into_broken
+                    //     can be called directly on Uninitielized timeline
+                    //     also leades to redundant .clone
+                    let broken_timeline = self
+                        .create_timeline_data(
+                            timeline_id,
+                            up_to_date_metadata.clone(),
+                            ancestor,
+                            None,
+                        )
+                        .with_context(|| {
+                            format!(
+                            "Failed to crate broken timeline data for {tenant_id}/{timeline_id}"
+                        )
+                        })?;
+                    broken_timeline.set_state(TimelineState::Broken);
+                    timelines_accessor.insert(timeline_id, Arc::new(broken_timeline));
+                    Err(e)
+                }
+            }
+        }?;
+
+        if self.remote_storage.is_some() {
+            // Reconcile local state with remote storage, downloading anything that's
+            // missing locally, and scheduling uploads for anything that's missing
+            // in remote storage.
+            timeline
+                .reconcile_with_remote(
+                    up_to_date_metadata,
+                    remote_startup_data.as_ref().map(|r| &r.index_part),
+                )
+                .await
+                .context("failed to reconcile with remote")?
+        }
+
+        // Save the metadata file to local disk.
+        if !picked_local {
+            save_metadata(
+                self.conf,
+                timeline_id,
+                tenant_id,
+                up_to_date_metadata,
+                first_save,
+            )
+            .context("save_metadata")?;
+        }
+
+        // Finally launch walreceiver
+        timeline.launch_wal_receiver();
+
+        Ok(())
+    }
+
+    ///
+    /// Attach a tenant that's available in cloud storage.
+    ///
+    /// This returns quickly, after just creating the in-memory object
+    /// Tenant struct and launching a background task to download
+    /// the remote index files.  On return, the tenant is most likely still in
+    /// Attaching state, and it will become Active once the background task
+    /// finishes. You can use wait_until_active() to wait for the task to
+    /// complete.
+    ///
+    pub fn spawn_attach(
+        conf: &'static PageServerConf,
+        tenant_id: TenantId,
+        remote_storage: &GenericRemoteStorage,
+    ) -> anyhow::Result<Arc<Tenant>> {
+        // FIXME: where to get tenant config when attaching? This will just fill in
+        // the defaults
```

Comment:
Honely, looking at `main` code, I don't get where we allocate the `Arc<Tenant>` during attach. Where do we do that? 

Looking at the only user of tenant attach in `cloud.git` code: https://github.com/neondatabase/cloud/blob/ef8b327bd044be52688fee9f192f817b768b967a/goapp/internal/controllers/managementapiv2/pageserver.go#L117-L145

Seems like we first attach the tenant, then issue a config RPC.

I'm confused.

Reply:
@hlinnaka and my conclusion: this is a bug in current console already: we're not migrating the tenant config. We should be doing that. => Follow up issue.

Reply:
Yep, it is a follow up even for main https://github.com/neondatabase/neon/issues/1555

Reply:
Will update the FIXME to point there.